napari_tiff/napari_tiff_reader.py

```diff
@@ -299,7 +313,27 @@ def imagej_reader(tif):
         blending=blending,
         visible=visible,
     )
-    return [(data, kwargs, 'image')]
+    return kwargs
+
+
+def get_ome_tiff_metadata(tif):
+    metadata = xml2dict(tif.ome_metadata)
+    if 'OME' in metadata:
+        metadata = metadata['OME']
+
+    # TODO: process ome metadata
```

Comment:
It might be worth depending on @tlambert03's [ome-types](https://github.com/tlambert03/ome-types) for this.

Reply:
I understood this is mainly used for formatting/writing as opposed to reading. We have a dictionary at this point so not sure what including the ome-types package would add here?

Reply:
ome-types does both, and inasmuch as you're only going to be pulling a couple things like channel names, colors, and scales from the metadata, I agree that it's probably not necessary to bring on an additional dependency just to grab a few things.

you will, however, likely eventually run into malformed xml that doesn't strictly meet the ome data model.  And in that case you'll just want to make sure that you essentially assume very little about the structure of the dict you get back (and fail gracefully).  ome-types tests itself against a ton of xml examples and might do slightly better in that regard, and the object you get back will have a guaranteed structure (and fixes a handful of commonly seen errors in various ome xml implementations, like micro-manager).  

I'd say go with what tifffile provides for now, and if you get any bug reports with poorly-mined metadata, then try out ome-types on those files and see if it makes things any more robust